homeassistant/components/ssdp/__init__.py

```diff
@@ -477,7 +477,7 @@ async def _ssdp_listener_callback(
 
```

Comment:
I think `_ssdp_listener_callback` needs to be a normal function to avoid the task creation in the lib

Something like https://github.com/bdraco/home-assistant/commit/73d1d4cf1e447ed35d3b8dd213076a1cfbacecd9

Reply:
Thank you for the suggestion, I've changed the callback. Can you do another review?

Reply:
Visual check looks good.  I'm going to install it on my dev system and test it.  If that goes well I'll put it on my production one

Reply:
Its much better now.  I still see some tasks being created but I haven't figured out where they are coming from or if we can get rid of them.... digging in now

![ssdp](https://user-images.githubusercontent.com/663432/208524628-d54b38d2-8669-40bc-b159-0cd187c4d7b0.png)

Reply:
`2022-12-19 11:16:47.131 WARNING (MainThread) [async_upnp_client.ssdp] Received packet from ('192.168.107.1', 41324): {'HOST': '239.255.255.250:1900', 'CACHE-CONTROL': 'max-age=60', 'LOCATION': 'http://192.168.107.1:37393/rootDesc.xml', 'SERVER': 'Debian/bullseye UPnP/1.1 MiniUPnPd/2.1', 'NT': 'urn:schemas-upnp-org:service:Layer3Forwarding:1', 'USN': 'uuid:aff51d27-0cd9-4094-838f-df4274b4f753::urn:schemas-upnp-org:service:Layer3Forwarding:1', 'NTS': 'ssdp:alive', 'OPT': '"http://schemas.upnp.org/upnp/1/0/"; ns=01', '01-NLS': '1670722186', 'BOOTID.UPNP.ORG': '1670722186', 'CONFIGID.UPNP.ORG': '1337', '_timestamp': datetime.datetime(2022, 12, 19, 11, 16, 47, 131584), '_host': '192.168.107.1', '_port': 41324, '_local_addr': ('0.0.0.0', 1900), '_remote_addr': ('192.168.107.1', 41324), '_udn': 'uuid:aff51d27-0cd9-4094-838f-df4274b4f753', '_location_original': 'http://192.168.107.1:37393/rootDesc.xml', 'location': 'http://192.168.107.1:37393/rootDesc.xml'} async_on_data=<bound method SsdpSearchResponder._async_on_data of <async_upnp_client.server.SsdpSearchResponder object at 0x10b16fa00>> on_data=None`

There is still a place that is using `async_on_data` in https://github.com/StevenLooman/async_upnp_client/pull/153/files#diff-164d6041a3b26e13c176d96d1c40bd16c2a65ff18c5c18b0cb63d8e31a227d30R259

Reply:
The remaining are coming from `SsdpSearchResponder`. I'm not 100% sure but looks like everything in there is coroutine but I think its possible to change them all to normal function as they all seem to wrap around `_async_send_response` which doesn't `await` anything.

Reply:
You are right! I've changed the functions to non-async functions. Can you do another test?

Reply:
Testing now

Reply:
Basic functional testing looks good. 

Doing some profiles now to look for performance any any unexpected results

Reply:
![tasks_gone](https://user-images.githubusercontent.com/663432/208758066-df3eed1e-2f35-42e9-8f46-d678c5dc34e2.png)

Tasks are gone.  Looks great 👍

Reply:
While the run time decrease is a great win, the decrease in latency is the bigger win since the time between when something uses `call_soon` and when its executed is going to be much shower since the `call_soon` queue is going to be much shorter.